sql/core/src/main/scala/org/apache/spark/sql/columnar/InMemoryColumnarTableScan.scala

```diff
@@ -314,7 +314,7 @@ private[sql] case class InMemoryColumnarTableScan(
                 columnAccessors(i).extractTo(nextRow, i)
                 i += 1
               }
-              nextRow
+              if (attributes.isEmpty) Row.empty else nextRow
```

Comment:
We need at least one column so that we know how many rows to produce, but we should not include it in the output otherwise we will not match the schema we are claiming to have.  It would be better to just remember the number of rows in a partition and use that instead.  This is just a quick fix so we don't return the wrong answer.